pkg/api/aim/runs.go

```diff
@@ -209,26 +208,22 @@ func GetRunMetrics(c *fiber.Ctx) error {
 		return fmt.Errorf("unable to find run %q: %w", p.ID, err)
 	}
 
+	subQuery := database.DB
+	for metricKey, metricContext := range metricKeysMap {
+		subQuery = subQuery.Or("key = ? AND json = ?", metricKey, metricContext)
+	}
+
 	// fetch run metrics based on provided criteria.
 	var data []database.Metric
-	if err := database.DB.Where(
-		"run_uuid = ?", p.ID,
-	).InnerJoins(
+	if err := database.DB.InnerJoins(
 		"Context",
-		func() *gorm.DB {
-			query := database.DB
-			for _, context := range contexts {
-				query = query.Or("json = ?", context)
-			}
-			return query
-		}(),
-	).Where(
-		"key IN ?", metricKeys,
```

Comment:
it looks like `metricKeys` is not used anymore, so there is no point in building it earlier in this function